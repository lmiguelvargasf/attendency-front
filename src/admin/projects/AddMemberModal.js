import React from 'react'
import { Space, Modal, Select } from 'antd'
const { Option } = Select

const AddMemberModal = ({ state, handleOk, handleCancel, handleOnChange }) => {
  return (
    <Modal
      title='Add Member'
      visible={state.visible}
      onOk={handleOk}
      confirmLoading={state.confirmLoading}
      onCancel={handleCancel}
    >
      {
        state.nonMembers.length > 0 ? (
          <Space>
            <strong>Member:</strong>
            <Select style={{ width: 150 }} onChange={handleOnChange} value={state.memberToAdd}>
              {state.nonMembers.map(nonMember => (
                <Option key={nonMember.key} value={nonMember.key}>
                  {nonMember.preferredName ? nonMember.preferredName : `${nonMember.firstName} ${nonMember.lastName}`}
                </Option>
              ))}
            </Select>
          </Space>
        ) : (
          <span>There are no more members to add to this project.</span>
        )
      }
    </Modal>
  )
}

export default AddMemberModal
