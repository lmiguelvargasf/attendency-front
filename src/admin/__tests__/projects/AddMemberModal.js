import React from 'react'
import { render, cleanup } from '@testing-library/react'
import AddMemberModal from '../../projects/AddMemberModal'

describe('AddMemberModal component', () => {
  afterEach(cleanup)

  describe('no members to be added', () => {
    let component
    beforeEach(() => {
      const state = {
        visible: true,
        confirmLoading: false,
        nonMembers: [],
        memberToAdd: null
      }
      component = (
        <AddMemberModal state={state} handleOk={jest.fn()} handleCancel={jest.fn()} handleOnChange={jest.fn()} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays text indicating no members to add', () => {
      const { getByTestId, queryByTestId } = render(component)
      expect(getByTestId('no-members-left')).toHaveTextContent('no more members to add')
      expect(queryByTestId('add-member-select')).toBeNull()
    })
  })

  describe('there are members to be added', () => {
    let component
    beforeEach(() => {
      const state = {
        visible: true,
        confirmLoading: false,
        nonMembers: [
          { key: 1, preferredName: 'Alpha' },
          { key: 1, preferredName: 'Beta' },
          { key: 1, preferredName: 'Gamma' }
        ],
        memberToAdd: null
      }
      component = (
        <AddMemberModal state={state} handleOk={jest.fn()} handleCancel={jest.fn()} handleOnChange={jest.fn()} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays text indicating no members to add', () => {
      const { getByTestId, queryByTestId } = render(component)
      expect(getByTestId('add-member-select')).not.toBeNull()
      expect(queryByTestId('no-members-left')).toBeNull()
    })
  })
})
