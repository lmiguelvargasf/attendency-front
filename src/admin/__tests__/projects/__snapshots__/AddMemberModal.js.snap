// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMemberModal component no members to be added matches snapshot 1`] = `<DocumentFragment />`;

exports[`AddMemberModal component there are members to be added matches snapshot 1`] = `<DocumentFragment />`;
